sdk/storage/inc/shares/protocol/share_rest_client.hpp

```diff
@@ -2557,18 +2558,41 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         {
           // Success
           ShareGetPropertiesResponse result;
-          result.Metadata = response.GetHeaders().at(Details::c_HeaderMetadata);
+
+          for (auto i = response.GetHeaders().lower_bound(Details::c_HeaderMetadata);
```

Comment:
`c_HeaderMetadata` should be defined as `x-ms-meta-`

Reply:
Swagger uses `"x-ms-meta"` as the name of metadata. I think it's better to align.